Misc/NEWS.d/next/C API/2022-07-29-19-35-30.gh-issue-95348.p5CDaH.rst

```diff
@@ -0,0 +1,4 @@
+Split ``clear_weakref`` into two functions: ``clear_weakref``
+and ``remove_weakref``. Abstracts ``remove_weakref`` with
+``remove_weakref_from_referent`` in weakrefobject.c.
+Patch by Robert O'Shea.

```

Comment:
I'm not sure that a NEWS entry is warranted.  If this PR changes any behavior then something has gone wrong, so there isn't anything to communicate to users.  This change is strictly about slightly improving the clarity of the Objects/weakrefobject.c code.  That sort of minor maintainability improvement typically does not get a NEWS entry.